pkg/controller/operators/olm/operator.go

```diff
@@ -1241,14 +1249,112 @@ func (a *Operator) syncCopyCSV(obj interface{}) (syncError error) {
 	}).Debug("copying csv to targets")
 
 	// Check if we need to do any copying / annotation for the operatorgroup
-	if err := a.ensureCSVsInNamespaces(clusterServiceVersion, operatorGroup, NewNamespaceSet(operatorGroup.Status.Namespaces)); err != nil {
-		logger.WithError(err).Info("couldn't copy CSV to target namespaces")
-		syncError = err
+	namespaceSet := NewNamespaceSet(operatorGroup.Status.Namespaces)
+	if a.copiedCSVs || !namespaceSet.IsAllNamespaces() {
```

Comment:
nit: +1 to Per's comment about renaming `copiedCSVs`. This line would read more clearly given that change.